lib/response.js

```diff
@@ -70,18 +70,43 @@ res.status = function status(code) {
  *      last: 'http://api.example.com/users?page=5'
  *    });
  *
- * @param {Object} links
+ *    res.links([{
+ *          href: 'http://api.example.com/users?page=2',
+ *          rel: 'next',
+ *          title: 'next chapter',
+ *          type: 'text/plain;charset=UTF-8'
+ *      }, {
+ *          href: 'http://api.example.com/users?page=5',
+ *          rel: 'last',
+ *          title: 'last chapter',
+ *          type: 'text/plain;charset=UTF-8'
+ *      }]);
+ *
+ *
+ * @param {Object|Array} links
  * @return {ServerResponse}
  * @public
  */
 
-res.links = function(links){
+res.links = function(links) {
   var link = this.get('Link') || '';
   if (link) link += ', ';
-  return this.set('Link', link + Object.keys(links).map(function(rel){
-    return '<' + links[rel] + '>; rel="' + rel + '"';
-  }).join(', '));
-};
+  if (Array.isArray(links)) { 
+    var attrTest = /^[a-z][a-z-*]+$/;
+    for (var i = 0; i < links.length; i++) { 
+      if (i !== 0) link += ', '; // do not prepend ', ' if it's the first link
+      link += '<' + links[i].href + '>; ';
+      delete links[i].href; // already appended
+      link += Object.keys(links[i]).filter(function (key) { return attrTest.test(key); })
+        .map(function(key) { return key + '="' + links[i][key] + '"'; }).join('; ');
```

Comment:
There are two valid parameters for this function, an object like:
```javascript
{
    next: 'http://api.example.com/users?page=2',
    last: 'http://api.example.com/users?page=5'
}
```
Which generates a link like this
```<http://api.example.com/users?page=2>; rel="next", <http://api.example.com/users?page=5>; rel="last", <http://api.example.com/users?page=1>; rel="prev"```

That's the implementation you see on line 106 and kept that way for backwards compatibility

This one is for the array parameter, which lets the user enter fine tune the links header.
What we can do here is to make no assumptions about the attribute value and let the user define the attributes like this:

```javascript
{
    href: '"http://api.example.com/users?page=2"', // Quoted string
    rel: '"next"', // Quoted string
    hreflang: 'en' // Not quoted
}
```
Generating something like this:

```<http://api.example.com/users?page=2>; rel="next"; hreflang=en```

Thoughts?